lib/Component.lua

```diff
@@ -101,6 +106,14 @@ end
 	current state object.
 ]]
 function Component:setState(partialState)
+	-- State cannot be set in any of the following places:
+	-- * During the component's init function
+	-- * During the component's render function
+	-- * After the component has been unmounted (or is in the process of unmounting, e.g. willUnmount)
+	if not self._canSetState then
+		error("State cannot be set at this point: are you setting state from an init, render, or willUnmount function?", 0)
```

Comment:
It might be a little clearer to say `are you calling setState from` instead of `are you setting state from`, since you can you set state from `init` as long as you don't call `setState` itself!